pkgs/by-name/no/nodeinfo/package.nix

```diff
@@ -0,0 +1,27 @@
+{
+  lib,
+  fetchzip,
+  buildGoModule
+}:
+buildGoModule rec {
+  pname = "nodeinfo";
+  version = "0.3.2";
+  hash = "sha256-OH0KAhocSiO6Hx78V5Ff7kKPzzRgEX4eS6/QX20VbIE=";
+
+  src = fetchzip {
+    url = "https://codeberg.org/thefederationinfo/nodeinfo-go/archive/v${version}.zip";
+    hash = "${hash}";
+    stripRoot = true;
+  } + "cli";
```

Comment:
Use [`sourceRoot`](https://nixos.org/manual/nixpkgs/stable/#var-stdenv-sourceRoot) attribute to build from the repo subdir.